pandas/core/indexes/period.py

```diff
@@ -272,19 +272,13 @@ def _shallow_copy(self, values=None, **kwargs):
             values = self._data
 
         if isinstance(values, type(self)):
-            values = values._values
+            values = values._data
 
         if not isinstance(values, PeriodArray):
             if isinstance(values, np.ndarray) and is_integer_dtype(values.dtype):
                 values = PeriodArray(values, freq=self.freq)
             else:
-                # in particular, I would like to avoid period_array here.
-                # Some people seem to be calling use with unexpected types
-                # Index.difference -> ndarray[Period]
-                # DatetimelikeIndexOpsMixin.repeat -> ndarray[ordinal]
-                # I think that once all of Datetime* are EAs, we can simplify
-                # this quite a bit.
-                values = period_array(values, freq=self.freq)
+                raise TypeError(type(values), getattr(values, "dtype", None))
```

Comment:
what triggers this?

Reply:
After this PR, this is never reached.  In master we get there with ndarray[object] via dropna and symmetric_difference. The PeriodIndex set ops PRs over the last couple days have been aimed at this

Reply:
ok can u make a comment about this then 

should this just be an assertion ?